File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/Java/C/s893190600_comments_removed.c
#include <stdio.h>
#include <string.h>

int main() {
    int H, W;
    scanf("%d %d", &H, &W);
    char grid[H+2][W+2];
    memset(grid, 0, sizeof(grid));

    for (int i = 1; i <= H; i++) {
        scanf("%s", grid[i]+1);
    }

    if (H == 1 && W == 1) {
        printf("No\n");
        return 0;
    }

    for (int i = 1; i <= H; i++) {
        for (int j = 1; j <= W; j++) {
            if (grid[i][j] == '#') {
                if ( (grid[i-1][j] != '#') &&
                     (grid[i+1][j] != '#') &&
                     (grid[i][j-1] != '#') &&
                     (grid[i][j+1] != '#') ) {
                    printf("No\n");
                    return 0;
                }
            }
        }
    }
    printf("Yes\n");
    return 0;
}
// End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/Java/Go/s893190600_comments_removed.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	writer := bufio.NewWriter(os.Stdout)
	defer writer.Flush()

	var H, W int
	fmt.Fscanln(reader, &H, &W)

	grid := make([][]byte, H)
	for i := 0; i < H; i++ {
		line, _ := reader.ReadString('\n')
		line = strings.TrimSpace(line)
		grid[i] = []byte(line)
	}

	if H == 1 && W == 1 {
		fmt.Fprintln(writer, "No")
		return
	}

	for i := 0; i < H; i++ {
		for j := 0; j < W; j++ {
			if grid[i][j] == '#' {
				hasAdjacent := false
				if i-1 >= 0 && grid[i-1][j] == '#' {
					hasAdjacent = true
				}
				if i+1 < H && grid[i+1][j] == '#' {
					hasAdjacent = true
				}
				if j-1 >= 0 && grid[i][j-1] == '#' {
					hasAdjacent = true
				}
				if j+1 < W && grid[i][j+1] == '#' {
					hasAdjacent = true
				}
				if !hasAdjacent {
					fmt.Fprintln(writer, "No")
					return
				}
			}
		}
	}

	fmt.Fprintln(writer, "Yes")
}

// End of Code
